server/spec/job_status_spec.rb

```diff
@@ -161,64 +160,77 @@
   it 'should not allow user to cancel a job it did not initiate' do
     system = consumer_client(@user, 'system7')
     job = system.consume_product(@monitoring.id, { :async => true })
-    status = system.get_job(job['id'])
+    status = system.get_async_job(job['id'])
     lambda do
-      @user.cancel_job(job['id'])
+      @user.cancel_async_job(job['id'])
     end.should raise_exception(RestClient::Forbidden)
   end
 
-  it 'should not allow user to view job status outside of managed org' do
+  # TODO Reenable after we decide how to check for access from outside of managed org
+  xit 'should not allow user to view job status outside of managed org' do
     other_user = user_client(@owner2, random_string("other_user"))
     system = consumer_client(other_user, random_string("another_system"))
     job = system.consume_product(@monitoring.id, { :async => true })
     # wait for job to complete, or test clean up will conflict with the asynchronous job.
-    wait_for_job(job['id'], 15)
+    wait_for_async_job(job['id'], 15)
     lambda do
-      @user.get_job(job['id'])
+      @user.get_async_job(job['id'])
     end.should raise_exception(RestClient::Forbidden)
-
   end
 
   it 'should allow consumer to view status of own job' do
     system = consumer_client(@user, 'system7')
     job = system.consume_product(@monitoring.id, { :async => true })
-    status = system.get_job(job['id'])
+    status = system.get_async_job(job['id'])
     status['id'].should eq(job['id'])
     # wait for job to complete, or test clean up will conflict with the asynchronous job.
-    wait_for_job(status['id'], 15)
+    wait_for_async_job(status['id'], 15)
   end
 
   it 'should not allow consumer to access another consumers job status' do
     system1 = consumer_client(@user, 's1')
     system2 = consumer_client(@user, 's2')
 
     job = system1.consume_product(@monitoring.id, { :async => true })
-    status = system1.get_job(job['id'])
+    status = system1.get_async_job(job['id'])
     # wait for job to complete, or test clean up will conflict with the asynchronous job.
-    wait_for_job(status['id'], 15)
+    wait_for_async_job(status['id'], 15)
 
     lambda do
-      system2.get_job(job['id'])
+      system2.get_async_job(job['id'])
     end.should raise_exception(RestClient::Forbidden)
   end
 
   it 'should allow consumer to cancel own job' do
-    system = consumer_client(@user, 'system7')
+    @cp.set_async_scheduler_status(false)
+    system = consumer_client(@user, 'system7', :system,  nil,  {}, @owner['key'])
     job = system.consume_product(@monitoring.id, { :async => true })
-    status = system.get_job(job['id'])
-    system.cancel_job(job['id'])
+    status = system.get_async_job(job['id'])
+    system.cancel_async_job(job['id'])
     # wait for job to complete, or test clean up will conflict with the asynchronous job.
-    wait_for_job(status['id'], 15)
+    wait_for_async_job(status['id'], 15)
+    @cp.set_async_scheduler_status(true)
```

Comment:
This is dangerous. If this test doesn't complete for any reason, the scheduler will be left in the paused state and will impact many other tests.

If we absolutely need this functionality for the test, we should do something similar to how created entities are cleaned up automatically at the end of every test to ensure the scheduler is turned back on at the end of a test, even if it fails.

Reply:
Done. I added after(:each) to clean it up.